utils/split_inouts.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python3
+"""
+This script provides a temporary solution for the problem of inout top level
+port representation of the BLIF format.
+
+The script loads the design from a JSON file generated by Yosys. Then it splits
+all inout ports along with their nets and connections to cell ports into two.
+Suffixes are automatically added to distinguish between the input and the
+output part.
+
+For example in the given design (in verilog):
+
+ module top(
+  input  A,
+  output B,
+  inout  C
+ );
+
+  IOBUF iobuf (
+   .I(A),
+   .O(B),
+   .IO_$inp(C),
+   .IO_$out(C)
+  );
+
+ endmodule
+
+the resulting design would be:
+
+ module top(
+  input  A,
+  output B,
+  input  C_$inp,
+  output C_$out
+ );
+
+  IOBUF iobuf (
+   .I(A),
+   .O(B),
+   .IO_$inp(C_$inp),
+   .IO_$out(C_$out)
+  );
+
+ endmodule
+
+"""
+import argparse
+import os
+import simplejson as json
+
+# =============================================================================
+
+
+def find_top_module(design):
+    """
+    Looks for the top-level module in the design. Returns its name.
+    """
+
+    for name, module in design["modules"].items():
+        attrs = module["attributes"]
+        if "top" in attrs and attrs["top"] == 1:
+            return name
+
+    return None
+
+
+def get_nets(bits):
+    """
+    Returns a set of numbers corresponding to net indices.
+    """
+    return set([n for n in bits if isinstance(n, int)])
+
+
+def get_free_net(nets):
+    """
+    Given a set of used net indices, returns a new, free index.
+    """
+    sorted_nets = sorted(list(nets))
+
+    # Find a gap in the sequence
+    for i in range(len(nets) - 1):
+        n0 = sorted_nets[i]
+        n1 = sorted_nets[i + 1]
+        if n1 != (n0 + 1):
+            return n0 + 1
+
+    # No gap was found, return max + 1.
+    return sorted_nets[-1] + 1
+
+
+# =============================================================================
+
+
+def find_and_split_inout_ports(design, module_name):
+
+    # Get the module
+    module = design["modules"][module_name]
+
+    # Find indices of all used nets
```

Comment:
I think you need to document the income JSON format here in the comment (or at least link to other documentation)

Reply:
I linked up Yosys documentation to the docstring of the file. Couldn't find any "official" documentation of the JSON design format other than Yosys' manual for the `write_json` command.